src/mam4xx/generated_gas_chem.hpp

```diff
@@ -0,0 +1,346 @@
+#ifndef MAM4XX_PP_CHEMISTRY_HPP
+#define MAM4XX_PP_CHEMISTRY_HPP
+// pp_linoz_mam4_resus_mom_soag
+// Generated code.
+// Authors: Oscar Diaz-Ibarra (odiazib@sandia.gov)
+//          Mike Schmidt (mjschm@sandia.gov)
+namespace mam4 {
+namespace gas_chemistry {
+const int rxntot = 7;     // number of total reactions
+const int gas_pcnst = 31; // number of gas phase species
+const int nzcnt = 32;     // number of non-zero matrix entries
+const int clscnt4 = 30;   // number of species in implicit class
+const int extcnt = 9;     // number of species with external forcing
+const int nfs = 8;        // number of fixed species
+const int permute_4[gas_pcnst] = {0,  1,  2,  3,  4,  5,  6,  7,  8,  9,
+                                  10, 11, 12, 13, 14, 15, 16, 17, 18, 19,
+                                  20, 21, 22, 23, 24, 25, 26, 27, 28, 29};
+const int clsmap_4[gas_pcnst] = {1,  2,  3,  4,  5,  6,  7,  8,  9,  10,
+                                 11, 12, 13, 14, 15, 16, 17, 18, 19, 20,
+                                 21, 22, 23, 24, 25, 26, 27, 28, 29, 30};
+enum class SpeciesId {
+  O3 = 0,
+  H2O2 = 1,
+  H2SO4 = 2,
+  SO2 = 3,
+  DMS = 4,
+  SOAG = 5,
+  so4_a1 = 6,
+  pom_a1 = 7,
+  soa_a1 = 8,
+  bc_a1 = 9,
+  dst_a1 = 10,
+  ncl_a1 = 11,
+  mom_a1 = 12,
+  num_a1 = 13,
+  so4_a2 = 14,
+  soa_a2 = 15,
+  ncl_a2 = 16,
+  mom_a2 = 17,
+  num_a2 = 18,
+  dst_a3 = 19,
+  ncl_a3 = 20,
+  so4_a3 = 21,
+  bc_a3 = 22,
+  pom_a3 = 23,
+  soa_a3 = 24,
+  mom_a3 = 25,
+  num_a3 = 26,
+  pom_a4 = 27,
+  bc_a4 = 28,
+  mom_a4 = 29,
+  num_a4 = 30,
+}; // enum class GasId
```

Comment:
This looks great! If it's not difficult, it might be nice to leave a blank line between enums, functions, etc, to improve readability.

Reply:
Done.